### PR TITLE
Update the sameday and eachday filter to accept TW5 date strings.

### PR DESCRIPTION
**Pull request by roma0104**
Updated sameday and eachday filters to accept TW5 date strings in addition to Date objects for comparison.

### CODE REVIEW

**Comment by roma0104** [comment]
This resolves an issue for my project and the limitation is not outlined in the documentation. 
https://github.com/roma0104/gsd5/issues/46

**Comment by Jermolene** [comment]
Hi @roma0104 thanks for this. My first thought is whether we need to go further to provide consistent, universal support for ISO date fields. In particular, we could update `$tw.utils.parseDate()` to allow it to accept ISO date strings, and then call it from sameday.js and eachday.js. What do you think?

**Comment by roma0104** [comment]
@Jermolene Sorry Jeremy, I misspoke.  Not ISO Strings but date format used by TiddlyWiki5, which resembles ISO.  I store some data on user created tiddlers using the `YYYYMMDDHHMMSSmmm` format.  Modified and Created fields get converted to Date objects when the tiddler is requested, but not my custom fields, for obvious reasons.  This pull is meant to allow custom fields to use the sameday and eachday filter by passing in the TW5 date/time string; the filters only currently accept Date objects.

File: core/modules/filters/eachday.js [hunk 1]
```diff
@@ -24,9 +24,16 @@ exports.eachday = function(source,operator,options) {
...
+	var toDateObj = function(value) {
+		if(!(value instanceof Date)) {
+			value = new Date($tw.utils.parseDate(value));
+		}
+		return value;
+	};
 	source(function(tiddler,title) {
 		if(tiddler && tiddler.fields[fieldName]) {
-			var value = toDate(tiddler.fields[fieldName]);
+			var value = toDate(toDateObj(tiddler.fields[fieldName]));
```
Comment (Jermolene) [inline core/modules/filters/eachday.js]: I think we can remove the `toDateObj()` function and instead change this line to:

```
var value = toDate($tw.utils.parseDate(tiddler.fields[fieldName]));
```

The reason is that both `toDate()` and `isSameDay()` convert their argument to a date if necessary, and `$tw.utils.parseDate()` copes correctly with an argument that is already a date object.

Comment (roma0104) [inline core/modules/filters/eachday.js]: Ah I see my mistake there.  I will update with some new commits hopefully
later in the day.

On Wed, Mar 11, 2015 at 3:26 PM, Jeremy Ruston <notifications@github.com>
wrote:

> In core/modules/filters/eachday.js
> <https://github.com/Jermolene/TiddlyWiki5/pull/1520#discussion_r26257422>:
>
> >  	source(function(tiddler,title) {
> >  		if(tiddler && tiddler.fields[fieldName]) {
> > -			var value = toDate(tiddler.fields[fieldName]);
> > +			var value = toDate(toDateObj(tiddler.fields[fieldName]));
>
> I think we can remove the toDateObj() function and instead change this
> line to:
>
> var value = toDate($tw.utils.parseDate(tiddler.fields[fieldName]));
>
> The reason is that both toDate() and isSameDay() convert their argument
> to a date if necessary, and $tw.utils.parseDate() copes correctly with an
> argument that is already a date object.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Jermolene/TiddlyWiki5/pull/1520/files#r26257422>.
>



-- 
Roma Quintin Hicks

**Comment by roma0104** [comment]
@Jermolene Could you advise any action on this topic?  

I think the acceptance of TW5 date strings is a necessary improvement to the two filters in question.  I am not convinced that true ISO date acceptance is needed.  I think the TW5 date string format is straightforward and simple; since it is the de jure date format for TW5, I personally don't see any need to support ISO, yet.  However, I can update the parseDate function if you would prefer it to accept ISO 8601.

**Comment by Jermolene** [comment]
Hi @roma0104 apologies I missed your commit. This looks good; as you say, we can always later update `parseDate ()` to deal with ISO 8601 dates.

**Comment by roma0104** [comment]
@Jermolene No worries,  I appreciate your oversight and acceptance of the merge.  I just wasn't sure if you needed something more/else from me.

**Pull request merged**